src/uwtools/rocoto.py

```diff
@@ -182,13 +182,56 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
-        for key, block in config.items():
-            tag, _ = self._tag_name(key)
-            if tag == STR.taskdep:
-                self._set_attrs(SubElement(e, STR.taskdep), block)
+        for tag, block in config.items():
+            tag, _ = self._tag_name(tag)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, tag=tag, block=block)
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config={tag: block})
+            elif tag in [STR.streq, STR.strneq]:
+                self._add_task_dependency_strequality(e, tag=tag, block=block)
             else:
                 raise UWConfigError("Unhandled dependency type %s" % tag)
 
+    def _add_task_dependency_operand(self, e: Element, tag: str, block: dict) -> None:
+        """
+        :param e: The parent element to add the new element to.
+        :param tag: Configuration new element to add.
+        :param block: Configuration data for the tag.
+        """
+        self._set_attrs(SubElement(e, tag), block)
+
+    def _add_task_dependency_operator(self, e: Element, config: dict) -> None:
+        """
+        Add an operator  element to the <dependency>.
+
+        :param e: The parent element to add the new element to.
+        :param config: Configuration data for this element.
+        """
+        operands = (STR.datadep, STR.taskdep, STR.timedep)
+        operators = (STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor)
+        strequality = (STR.streq, STR.strneq)
+
+        for tag, block in config.items():
+            tag, _ = self._tag_name(tag)
+            if tag in operators:
+                e = SubElement(e, tag)
+                self._add_task_dependency_operator(e, config=block)
+            elif tag in operands:
+                self._add_task_dependency_operand(e, tag=tag, block=block)
+            elif tag in strequality:
+                self._add_task_dependency_strequality(e, tag=tag, block=block)
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
```

Comment:
I think it's probably a good idea to keep this check here, but just to be sure: We think we're handling all dependency types as of this PR, right? Our schema checks should ensure that we don't get invalid inputs, but a final check never hurts.

Reply:
I added this check here and test case for raising an error as well. We should now be handling all dependency types. Happy to remove if we decide that our schema should catch this instead. I added a test case for this to confirm 100% coverage.

Reply:
I think it's good to keep it. Maybe someday we can arrange things so we're confident that all config data passes through schema validation but, for example, and at the moment, users can monkey with the content of `Config` objects by accessing its `data` member directly, which would bypass schema validation.